examples/energy-management-app/energy-management-common/include/EnergyEvseDelegateImpl.h

```diff
@@ -139,6 +141,46 @@ class EnergyEvseDelegate : public EnergyEvse::Delegate
      */
     Status StartDiagnostics() override;
 
+    /**
+     * @brief    Called when EVSE cluster receives the SetTargets command
+     */
+    Status SetTargets(
+        const DataModel::DecodableList<Structs::ChargingTargetScheduleStruct::DecodableType> & chargingTargetSchedules) override;
+
+    EvseTargetsDelegate * GetEvseTargetsDelegate() { return mEvseTargetsDelegate; }
+
+    /**
+     * @brief Delegate should implement a handler to PrepareGetTargets
+     *
+     * This needs to load any stored targets into memory and hold it until the GetTargetsFinished is
+     * called by the cluster server.
+     *
+     * @param  Reference to EvseTargetIterator class that implements the ability
+     *         for the cluster server to iterate through the target entries
+     */
+    CHIP_ERROR PrepareGetTargets(EvseTargetIterator ** iterator) override;
+
+    /**
+     * @brief Delegate should implement a handler to GetTargetsFinished
+     *
+     * This is used by the cluster server to indicate it has finished preparing
+     * the GetTargetsResponse using the iterator and that the memory in the delegate can
+     * freed
+     */
+    CHIP_ERROR GetTargetsFinished() override;
+
+    /**
+     * @brief    Called when EVSE cluster receives ClearTargets command
+     */
+    Status ClearTargets() override;
+
+    /* Helper functions for managing targets*/
```

Comment:
These helpers should probably not be public. Do they need to be?

Reply:
These are called by the src/app/ sdk layer so I think they need to be public.